backend/src/game/game.service.ts

```diff
@@ -23,30 +25,298 @@ export class GameService {
+    }
+    return pausingPlayer;
+  }
+
+  /****************************************************************************/
+  // find, create, join game
+  /****************************************************************************/
+  async getGameByID(gameID: string): Promise<Game> {
+    let game: Game;
+    const gameString: string = await this.cacheManager.get(gameID);
+    if (gameString && gameString != '') game = JSON.parse(gameString);
+    return game;
+  }
+
+  async findPausedGame(client: Socket): Promise<string> {
+    let gameID: string;
+    const user: User = await this.getSocketUser(client);
+    if (!user) {
+      return '';
+    }
+    const pausingPlayer: string = await this.cacheManager.get(
+      `game${user.email}`,
+    );
+    if (pausingPlayer) {
+      let pausingPlayerObject: Player = JSON.parse(pausingPlayer);
+      if (pausingPlayerObject.gameID != 'non') {
+        pausingPlayerObject.socketID = client.id;
+        await this.cacheManager.set(
+          `game${user.email}`,
+          JSON.stringify(pausingPlayerObject),
+        );
+        console.log(pausingPlayerObject);
+        console.log(
+          'Socket: existing player updated: ',
+          pausingPlayerObject.email,
+        );
+        gameID = pausingPlayerObject.gameID;
+        this.joinGameAndLaunch(pausingPlayerObject, pausingPlayerObject.gameID);
+      }
+    }
+    console.log("has paused game: ", gameID);
+    return gameID;
+  }
+
+  async createGame(client: Socket): Promise<Game> {
+    let newGame: Game;
     let pendingPlayer: string = await this.cacheManager.get('pendingPlayer');
+    // continue to joinGame if there is a pendingPlayer
+    let pausedGameID: string = await this.findPausedGame(client);
+    console.log("paused game in create game: ", pausedGameID);
+    if (!pendingPlayer && !pausedGameID) {
+      let player: Player = await this.getSocketPlayer(client);
+      if (player) {
+        console.log('create a pending game');
+        newGame = await this.createWaitingGame(player);
+      }
+    }
+    console.log("game CREATEEEDDDD!!!!!!");
+    return newGame;
+  }
 
+  /****************************************************************************/
+  // INVITE
+  /****************************************************************************/
+  async checkInvitedUserStatus(client: Socket, userEmail: string): Promise<string> {
+    let result: string;
+    const invitingUser: User = await this.getSocketUser(client);
+    const invitedUser: User = await this.getUserByEmail(userEmail);
+    if (invitedUser && invitingUser) {
+      if (invitedUser.email === invitingUser.email) {
+        result = 'Can not invite your self.';
+      } else if (invitedUser.status === 'ONLINE') {
+        result = invitedUser.status;
+      } else {
+        result = 'User not available.';
+      }
+    } else {
+      result = 'User not found.';
+    }
+    return result;
+  }
+
+  async createInvitingGame(player: Player): Promise<Game> {
```

Comment:
This function is identical to `createWaitingGame()`. They have the exact the same functionality (since the pending player is being handled elsewhere).

Reply:
It's really just a COPY XD. The only difference is "pendingGame" or "invite${player.email}" to be stored into cache. Will try to find some way to simplify things latter.